library_generation/utilities.sh

```diff
@@ -0,0 +1,170 @@
+  if [ "${rest_numeric_enums}" == "true" ]; then
+    rest_numeric_enums="rest-numeric-enums,"
+  else
+    rest_numeric_enums=""
+  fi
+  echo "transport=$transport,${rest_numeric_enums}grpc-service-config=$grpc_service_config,${gapic_config}api-service-config=$api_service_config"
+}
+
+remove_grpc_version() {
+  find "$destination_path" -type f -name "*Grpc.java" -exec \
+  sed -i 's/value = \"by gRPC proto compiler.*/value = \"by gRPC proto compiler\",/g' {} \;
+}
+
+download_gapic_generator_pom_parent() {
+  gapic_generator_version=$1
+  cd "$working_directory"
+  if [ ! -f "gapic-generator-java-pom-parent-$gapic_generator_version.pom" ]; then
+    mvn org.apache.maven.plugins:maven-dependency-plugin:copy -q \
+      -Dartifact=com.google.api:gapic-generator-java-pom-parent:"$gapic_generator_version":pom \
+      -DoutputDirectory="$working_directory" || \
+      download_fail "gapic-generator-java-pom-parent-$gapic_generator_version.pom"
+  fi
+}
+
+get_grpc_version() {
+  gapic_generator_version=$1
+  # get grpc version from gapic-generator-java-pom-parent/pom.xml
+  download_gapic_generator_pom_parent "$gapic_generator_version"
+  grpc_version=$(grep grpc.version "gapic-generator-java-pom-parent-$gapic_generator_version.pom" | sed 's/<grpc\.version>\(.*\)<\/grpc\.version>/\1/' | sed 's/^[[:space:]]*//;s/[[:space:]]*$//')
+  echo "$grpc_version"
+}
+
+get_protobuf_version() {
+  gapic_generator_version=$1
+  # get protobuf version from gapic-generator-java-pom-parent/pom.xml
+  download_gapic_generator_pom_parent "$gapic_generator_version"
+  protobuf_version=$(grep protobuf.version "gapic-generator-java-pom-parent-$gapic_generator_version.pom" | sed 's/<protobuf\.version>\(.*\)<\/protobuf\.version>/\1/' | sed 's/^[[:space:]]*//;s/[[:space:]]*$//' | cut -d "." -f2-)
+  echo "$protobuf_version"
+}
+
+download_tools() {
+  gapic_generator_version=$1
+  protobuf_version=$2
+  grpc_version=$3
+  download_generator "$gapic_generator_version"
+  download_protobuf "$protobuf_version"
+  download_grpc_plugin "$grpc_version"
+}
+
+download_generator() {
+  gapic_generator_version=$1
+  cd "$working_directory"
+  if [ ! -f "gapic-generator-java-$gapic_generator_version.jar" ]; then
+    mvn org.apache.maven.plugins:maven-dependency-plugin:copy -q \
+      -Dartifact=com.google.api:gapic-generator-java:"$gapic_generator_version" \
+      -DoutputDirectory="$working_directory" || \
+      download_fail "gapic-generator-java-$gapic_generator_version.jar"
+  fi
+}
+
+download_protobuf() {
+  protobuf_version=$1
+  cd "$working_directory"
+  if [ ! -d "protobuf-$protobuf_version" ]; then
+    # pull proto files and protoc from protobuf repository
+    # maven central doesn't have proto files
+    curl -LJ -o "protobuf-$protobuf_version.zip" https://github.com/protocolbuffers/protobuf/releases/download/v"$protobuf_version"/protoc-"$protobuf_version"-linux-x86_64.zip \
```

Comment:
@blakeli0 I still need to download protobuf from [protobuf releases](https://github.com/protocolbuffers/protobuf/releases) because there're common protos in the .zip file. These are the protos in protobuf.zip. Does `sdk-platform-java` publish these protos so that I can download these files from maven central as well?
```
-rw-r--r--  1 joewa primarygroup  6049 Aug 18 19:50 any.proto
-rw-r--r--  1 joewa primarygroup  7728 Aug 18 19:50 api.proto
drwxr-xr-x  2 joewa primarygroup  4096 Aug 18 19:29 compiler/
-rw-r--r--  1 joewa primarygroup 43479 Aug 18 19:50 descriptor.proto
-rw-r--r--  1 joewa primarygroup  4892 Aug 18 19:50 duration.proto
-rw-r--r--  1 joewa primarygroup  2363 Aug 18 19:50 empty.proto
-rw-r--r--  1 joewa primarygroup  8185 Aug 18 19:50 field_mask.proto
-rw-r--r--  1 joewa primarygroup  2341 Aug 18 19:50 source_context.proto
-rw-r--r--  1 joewa primarygroup  3778 Aug 18 19:50 struct.proto
-rw-r--r--  1 joewa primarygroup  6449 Aug 18 19:50 timestamp.proto
-rw-r--r--  1 joewa primarygroup  6367 Aug 18 19:50 type.proto
-rw-r--r--  1 joewa primarygroup  4044 Aug 18 19:50 wrappers.proto
```

Reply:
We might be able to download these files from Maven central somewhere but I don't think we can get away from downloading protoc, so this look good to me.